### DISCUSSION TITLE
NTLM Authentication

### DISCUSSION THREAD

**Discussion by espy19** [basdijkstra/rest-assured-net#129] [Ideas]
Hello. Just like `BasicAuth()` and `OAuth2()`, can we have something like `Ntlm()` so that the HTTP request uses the credentials of the current user? I can achieve this currently by passing my own `HttpClient` object (created using `Credentials = CredentialCache.DefaultNetworkCredentials`) but the `Ntlm()` wrapper could make it even easier.

**Comment by basdijkstra** [comment]
Hey @espy19, that sounds like a great idea.

What I think I would do is add two methods:
* One without arguments (so `Ntlm()`) that uses `CredentialCache.DefaultNetworkCredentials`
* One where you can supply the username, password and domain yourself, which would call [this](https://learn.microsoft.com/en-us/dotnet/api/system.net.networkcredential.-ctor?view=net-8.0#system-net-networkcredential-ctor(system-string-system-string-system-string)) `NetworkCredential` constructor

Does that sound like a plan?

**Reply by espy19** [reply]
Perfect!

**Reply by basdijkstra** [reply]
Good. Converted this to #130 and will start working on it soon. I'll let you know when I've got a beta release available so you can run some tests with it before I release it officially.

**Comment by basdijkstra** [comment]
FYI: https://github.com/basdijkstra/rest-assured-net/issues/130#issuecomment-2424108652
